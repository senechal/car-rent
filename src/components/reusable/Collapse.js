import PropTypes from 'prop-types';
import { DownOutlined, UpOutlined } from '@ant-design/icons';
import { useState, useCallback } from 'react';
import CollapsibleDiv from 'react-collapse';

const Collapse = ({ header, children, ...props }) => {
  const [collapse, setCollapse] = useState(false);
  const toggleCollapse = useCallback(() => {
    setCollapse(curr => !curr);
  }, [setCollapse]);

  return (
    <div {...props}>
      <div className="rac-collapse-header" onClick={toggleCollapse} data-testid="collapse">
        {header}
        {
          collapse
           ? <DownOutlined className="rac-collapse-indicator" />
           : <UpOutlined className="rac-collapse-indicator" />
        }
      </div>
      <CollapsibleDiv isOpened={!collapse} theme={{collapse: 'rac-collapse-body', content: 'rac-collapse-content'}} >
        {children}
      </CollapsibleDiv>
    </div>
  );
};

Collapse.propTypes = {
  header: PropTypes.string.isRequired,
  children: PropTypes.any,
};
Collapse.defaultProps = {
  children: '',
};

export default Collapse;
