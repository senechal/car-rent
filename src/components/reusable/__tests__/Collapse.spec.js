import { render, fireEvent } from '@testing-library/react';
import Collapse from '../Collapse';

describe('Testing Collapse', () => {
  it('should render correctly', () => {
    const { container } = render(<Collapse header="collapse">content</Collapse>);
    expect(container).toMatchSnapshot();
  });
  it('should call onClose when clicked', () => {
    const { getByTestId, container } = render(<Collapse header="collapse">content</Collapse>);
    const collapse = getByTestId('collapse');
    fireEvent.click(collapse);
    expect(container).toMatchSnapshot();
  });
});