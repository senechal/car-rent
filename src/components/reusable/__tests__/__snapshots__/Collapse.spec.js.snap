// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Collapse should call onClose when clicked 1`] = `
<div>
  <div>
    <div
      class="rac-collapse-header"
      data-testid="collapse"
    >
      collapse
      <span
        aria-label="down"
        class="anticon anticon-down rac-collapse-indicator"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="down"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </span>
    </div>
    <div
      aria-hidden="true"
      class="rac-collapse-body"
      style="height: 0px; overflow: hidden;"
    >
      <div
        class="rac-collapse-content"
      >
        content
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing Collapse should render correctly 1`] = `
<div>
  <div>
    <div
      class="rac-collapse-header"
      data-testid="collapse"
    >
      collapse
      <span
        aria-label="up"
        class="anticon anticon-up rac-collapse-indicator"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="up"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
          />
        </svg>
      </span>
    </div>
    <div
      aria-hidden="false"
      class="rac-collapse-body"
      style="height: auto; overflow: initial;"
    >
      <div
        class="rac-collapse-content"
      >
        content
      </div>
    </div>
  </div>
</div>
`;
